packages/plugin-hardhat/src/force-import.ts

```diff
@@ -108,5 +110,14 @@ async function addAdminToManifest(
   opts: ForceImportOptions,
 ) {
   const adminAddress = await getAdminAddress(provider, proxyAddress);
+  if (isEmptySlot(adminAddress)) {
+    // Asser that the admin slot of a transparent proxy is not zero, otherwise the simulation below would fail due to no code at the address.
+    // Note: Transparent proxies should not have the zero address as the admin, according to TransparentUpgradeableProxy's _setAdmin function.
```

Comment:
```suggestion
    // Note: Since contracts v5.0, transparent proxies should not have the zero address as the admin, according to TransparentUpgradeableProxy's _setAdmin function.
```

Reply:
This is also a restriction in 4.x though: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.9.3/contracts/proxy/ERC1967/ERC1967Upgrade.sol#L105

Reply:
Ah I see, I thought it was because the admin is now immutable in Transparent Proxy, now I see this wasn't just reached in the past.

Solving